packages/@glimmer/core/src/template.ts

```diff
@@ -1,92 +1,25 @@
 import { DEBUG } from '@glimmer/env';
-import {
-  SerializedTemplateWithLazyBlock,
-  Dict,
-  SerializedTemplateBlockJSON,
-} from '@glimmer/interfaces';
-
-export interface TemplateMeta {
-  scope: () => Dict<unknown>;
-}
-export type TemplateScope = () => Dict<unknown>;
-
-const TEMPLATE_MAP = new WeakMap<object, SerializedTemplateWithLazyBlock>();
-const getPrototypeOf = Object.getPrototypeOf;
-
-export const SCOPE_MAP = new WeakMap<SerializedTemplateWithLazyBlock, TemplateScope>();
-
-export interface CustomSerializedTemplate {
-  id?: string | null;
-  block: SerializedTemplateBlockJSON;
-  meta: TemplateMeta;
-}
+import { setComponentTemplate as vmSetComponentTemplate } from '@glimmer/manager';
+import { SerializedTemplateWithLazyBlock, Dict } from '@glimmer/interfaces';
+import { templateFactory } from '@glimmer/opcode-compiler';
 
 // This is provided by the `babel-plugin-strict-template-precompile` plugin
 export let createTemplate: (
   scopeOrTemplate: Dict<unknown> | string,
   template?: string
-) => CustomSerializedTemplate;
+) => SerializedTemplateWithLazyBlock;
 
 if (DEBUG) {
-  createTemplate = (): CustomSerializedTemplate => {
+  createTemplate = (): SerializedTemplateWithLazyBlock => {
```

Comment:
We should probably rename/alias this to `precompileTemplate`, to match the original RFC. We can do that in a follow on, just noting for the 2.0 release.

Reply:
https://github.com/glimmerjs/glimmer.js/issues/325